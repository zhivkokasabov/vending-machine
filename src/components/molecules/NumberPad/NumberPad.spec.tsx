import { TooltipProvider } from "@radix-ui/react-tooltip";
import { NumberPad } from ".";

import { fireEvent, render } from "@testing-library/react";

import * as codeProvider from "providers/Code.provider";

jest.mock("providers/Code.provider");

describe("NumberPad tests", () => {
  const appendMock = jest.fn();
  const onCashOutMock = jest.fn();
  const onCancelMock = jest.fn();
  const onConfirmMock = jest.fn();

  const renderComponent = () => render(
    <TooltipProvider>
      <NumberPad onCancel={onCancelMock} onCashOut={onCashOutMock} onConfirm={onConfirmMock} />
    </TooltipProvider>
  );

  beforeEach(() => {
      jest.spyOn(codeProvider, 'useCode').mockReturnValue({
        append: appendMock,
        resetCode: jest.fn(),
        code: ""
      });
    });

  it("should render", () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should propagate cancel", () => {
    const { getAllByRole } = renderComponent();

    fireEvent.click(getAllByRole("button")[9]);

    expect(onCancelMock).toHaveBeenCalledTimes(1);
  });

  it("should propagate confirm", () => {
    const { getAllByRole } = renderComponent();

    fireEvent.click(getAllByRole("button")[12]);

    expect(onConfirmMock).toHaveBeenCalledTimes(1);
  });

  it("should propagate cashOut", () => {
    const { getAllByRole } = renderComponent();

    fireEvent.click(getAllByRole("button")[11]);

    expect(onCashOutMock).toHaveBeenCalledTimes(1);
  });

  it("should call append with the selected character", () => {
    const { getAllByRole } = renderComponent();

    fireEvent.click(getAllByRole("button")[2]);
    fireEvent.click(getAllByRole("button")[3]);
    fireEvent.click(getAllByRole("button")[10]);

    expect(appendMock).toHaveBeenCalledTimes(3);
    expect(appendMock).toHaveBeenCalledWith(3);
    expect(appendMock).toHaveBeenCalledWith(4);
    expect(appendMock).toHaveBeenCalledWith(0);
  });
});
