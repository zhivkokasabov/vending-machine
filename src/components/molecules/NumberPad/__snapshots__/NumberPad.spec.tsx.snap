// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberPad tests should render 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-3 gap-2 mb-3"
  >
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      1
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      2
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      3
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      4
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      5
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      6
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      7
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      8
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      9
    </button>
    <span
      data-state="closed"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 bg-red-500 text-neutral-50 shadow-sm hover:bg-red-500/90 dark:bg-red-900 dark:text-neutral-50 dark:hover:bg-red-900/90 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
      >
        <div />
      </button>
    </span>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 border border-neutral-200 bg-white shadow-sm hover:bg-neutral-100 hover:text-neutral-900 dark:border-neutral-800 dark:bg-neutral-950 dark:hover:bg-neutral-800 dark:hover:text-neutral-50 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
    >
      0
    </button>
    <span
      data-state="closed"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 bg-neutral-900 text-neutral-50 shadow hover:bg-neutral-900/90 dark:bg-neutral-50 dark:text-neutral-900 dark:hover:bg-neutral-50/90 px-4 py-2 undefined rounded-full w-[3rem] h-[3rem]"
      >
        <div />
      </button>
    </span>
  </div>
  <button
    aria-label="purchase specified product"
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-950 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:focus-visible:ring-neutral-300 bg-neutral-900 text-neutral-50 shadow hover:bg-neutral-900/90 dark:bg-neutral-50 dark:text-neutral-900 dark:hover:bg-neutral-50/90 h-9 px-4 py-2 w-full"
  >
    <div />
     Purchase
  </button>
</DocumentFragment>
`;
